Framework/API/test/WorkspacePropertyWithIndexTest.h

```diff
@@ -0,0 +1,351 @@
+#ifndef MANTID_API_WORKSPACEPROPERTYWITHINDEXTEST_H_
+#define MANTID_API_WORKSPACEPROPERTYWITHINDEXTEST_H_
+
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidTestHelpers/FakeObjects.h"
+#include <boost/shared_ptr.hpp>
+#include <cxxtest/TestSuite.h>
+
+#include "MantidAPI/WorkspacePropertyWithIndex.tcc"
+#include "MantidKernel/PropertyManager.h"
+#include "MantidKernel/make_unique.h"
+#include <numeric>
+
+using namespace Mantid::API;
+using Mantid::API::WorkspacePropertyWithIndex;
+
+namespace {
+class PropertyManagerHelper : public PropertyManager {
+public:
+  PropertyManagerHelper() : PropertyManager() {}
+
+  using PropertyManager::declareProperty;
+  using PropertyManager::setProperty;
+  using PropertyManager::getPointerToProperty;
+};
+} // namespace
+
+using MatrixWorkspaceIndexProp = WorkspacePropertyWithIndex<MatrixWorkspace>;
+class WorkspacePropertyWithIndexTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static WorkspacePropertyWithIndexTest *createSuite() {
+    return new WorkspacePropertyWithIndexTest();
+  }
+  static void destroySuite(WorkspacePropertyWithIndexTest *suite) {
+    delete suite;
+  }
+
+  void testConstructor() {
```

Comment:
I think there are some simpler tests that aren't implemented here around object behaviour. For example, `operator==` is overloaded, but I don't see an explicit test for that. Would be good to test that `operator!=` does what you expect too.